pyomo/solvers/plugins/solvers/scip_direct.py

```diff
@@ -0,0 +1,769 @@
+        self._vars_referenced_by_con[con] = referenced_vars
+        self._pyomo_con_to_solver_con_expr_map[con] = scip_cons
+        self._pyomo_con_to_solver_con_map[con] = scip_cons.name
+        self._solver_con_to_pyomo_con_map[conname] = con
+
+    def _add_sos_constraint(self, con):
+        if not con.active:
+            return None
+
+        conname = self._symbol_map.getSymbol(con, self._labeler)
+        level = con.level
+        if level not in [1, 2]:
+            raise ValueError(f"Solver does not support SOS level {level} constraints")
+
+        scip_vars = []
+        weights = []
+
+        self._vars_referenced_by_con[con] = ComponentSet()
+
+        if hasattr(con, "get_items"):
+            # aml sos constraint
+            sos_items = list(con.get_items())
+        else:
+            # kernel sos constraint
+            sos_items = list(con.items())
+
+        for v, w in sos_items:
+            self._vars_referenced_by_con[con].add(v)
+            scip_vars.append(self._pyomo_var_to_solver_var_expr_map[v])
+            self._referenced_variables[v] += 1
+            weights.append(w)
+
+        if level == 1:
+            scip_cons = self._solver_model.addConsSOS1(
+                scip_vars, weights=weights, name=conname
+            )
+        else:
+            scip_cons = self._solver_model.addConsSOS2(
+                scip_vars, weights=weights, name=conname
+            )
+        self._pyomo_con_to_solver_con_expr_map[con] = scip_cons
+        self._pyomo_con_to_solver_con_map[con] = scip_cons.name
+        self._solver_con_to_pyomo_con_map[conname] = con
+
+    def _scip_vtype_from_var(self, var):
+        """
+        This function takes a pyomo variable and returns the appropriate SCIP variable type
+
+        Parameters
+        ----------
+        var: pyomo.core.base.var.Var
+            The pyomo variable that we want to retrieve the SCIP vtype of
+
+        Returns
+        -------
+        vtype: str
+            B for Binary, I for Integer, or C for Continuous
+        """
+        if var.is_binary():
+            vtype = "B"
+        elif var.is_integer():
+            vtype = "I"
+        elif var.is_continuous():
+            vtype = "C"
+        else:
+            raise ValueError(f"Variable domain type is not recognized for {var.domain}")
+        return vtype
+
+    def _set_objective(self, obj):
+        if self._objective is not None:
+            for var in self._vars_referenced_by_obj:
+                self._referenced_variables[var] -= 1
+            self._vars_referenced_by_obj = ComponentSet()
+            self._objective = None
+
+        if obj.active is False:
+            raise ValueError("Cannot add inactive objective to solver.")
+
+        if obj.sense == minimize:
+            sense = "minimize"
+        elif obj.sense == maximize:
+            sense = "maximize"
+        else:
+            raise ValueError(f"Objective sense is not recognized: {obj.sense}")
```

Comment:
I would instead do:
```
if obj.sense not in ['maximize', 'minimize']:
    raise ValueError(f"Objective sense is not recognized: {obj.sense}")
...
self._solver_model.setObjective(scip_expr, sense=obj.sense)
```

Reply:
This does not work because you're comparing to string objects. `maximize` and `minimzie` in the code above are imports from pyomo.

I think I could change it to `obj.sense.value` or `obj.sense.name` (can't remember), but I wasn't 100% confident in covering all cases. I've left it as is.